core/src/main/java/io/grpc/internal/ManagedChannelImpl.java

```diff
@@ -316,7 +215,7 @@ public ClientTransport get(CallOptions callOptions) {
       LoadBalancer.Factory loadBalancerFactory, ClientTransportFactory transportFactory,
       DecompressorRegistry decompressorRegistry, CompressorRegistry compressorRegistry,
       SharedResourceHolder.Resource<ScheduledExecutorService> timerService,
-      Supplier<Stopwatch> stopwatchSupplier, long idleTimeoutMillis,
+      Supplier<Stopwatch> stopwatchSupplier, Long idleTimeoutMillis,
```

Comment:
To simplify life I would recommend just creating a custom Optional class that does not conflict with Java 8, but allows it for Java 6.  Then you can use methods like `isEmpty()` or variations of it.